lib/apiGateway/schema.js

```diff
@@ -119,6 +129,71 @@ const partitionKey = Joi.alternatives().try([
     )
 ])
 
+const allowedDynamodbActions = ['PutItem', 'GetItem', 'DeleteItem', 'UpdateItem']
+const dynamodbDefaultKeyScheme = Joi.alternatives().try([
+  Joi.string(),
+  Joi.object()
+    .keys({
+      pathParam: Joi.string(),
+      queryStringParam: Joi.string(),
+      attributeType: Joi.string().required()
+    })
+    .xor('pathParam', 'queryStringParam')
+    .error(
+      customErrorBuilder(
+        'object.xor',
+        'key must contain "pathParam" or "queryStringParam" and only one'
+      )
+    )
+])
+const hashKey = Joi.when('action', {
+  is: Joi.string()
+    .valid(allowedDynamodbActions)
+    .required(),
+  then: dynamodbDefaultKeyScheme,
+  otherwise: Joi.when('method', {
+    is: Joi.string()
+      .valid('post')
+      .insensitive()
+      .required(),
+    then: Joi.string()
+      .required()
+      .error(
+        customMultipleErrorBuilder({
+          'string.base': [
+            '"hashKey" must be a string when you define post "method" and not define',
+            ' "action" expolicitly since the hashKey value is auto-generated on AWS end'
```

Comment:
I don't understand this error message... are we saying that, if `method` is `post` then you can't define `action`, and `hashKey` must be a string? Even then I'm still confused, what does the string value represent? the attribute name?

Reply:
also, when do you see this error message?